src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -1983,4 +1983,70 @@ public FileDragDropPreferenceType getEntryEditorFileLinkPreference() {
     public void storeEntryEditorFileLinkPreference(FileDragDropPreferenceType type) {
         put(ENTRY_EDITOR_DRAG_DROP_PREFERENCE_TYPE, type.name());
     }
+
+    @Override
+    public List<TemplateExporter> getCustomExportFormats(JournalAbbreviationLoader loader) {
+        int i = 0;
+        List<TemplateExporter> formats = new ArrayList<>();
+        String exporterName;
+        String filename;
+        String extension;
+        LayoutFormatterPreferences layoutPreferences = getLayoutFormatterPreferences(loader);
+        SavePreferences savePreferences = loadForExportFromPreferences();
+        List<String> s;
+        // possibly check if CUSTOM_EXPORT_FORMAT + 0 is empty too and throw error
+        while (!((s = getStringList(CUSTOM_EXPORT_FORMAT + i)).isEmpty())) {
+            exporterName = s.get(0);
```

Comment:
Pleas use constants for the indicies, e.g.  EXPORTER_NAME_INDEX = 0
And please use a better name for "s"

Reply:
I have fixed these issues.